concrete-core/src/specification/engines/glwe_ciphertext_ggsw_ciphertext_external_product.rs

```diff
@@ -0,0 +1,72 @@
+use crate::specification::engines::AbstractEngine;
+use crate::specification::entities::{GgswCiphertextEntity, GlweCiphertextEntity};
+
+use super::engine_error;
+
+engine_error! {
+    GlweCiphertextGgswCiphertextExternalProductError for
+    GlweCiphertextGgswCiphertextExternalProductEngine @
+    PolynomialSizeMismatch => "The GGSW ciphertext and GLWE ciphertext polynomial sizes must be the same.",
+    GlweDimensionMismatch => "The GGSW ciphertext and GLWE ciphertext GLWE dimension must be the same."
+}
+
+impl<EngineError: std::error::Error> GlweCiphertextGgswCiphertextExternalProductError<EngineError> {
+    /// Validates the inputs
+    pub fn perform_generic_checks<GlweCiphertext, GgswCiphertext>(
+        glwe_input: &GlweCiphertext,
+        ggsw_input: &GgswCiphertext,
+    ) -> Result<(), Self>
+    where
+        GlweCiphertext: GlweCiphertextEntity,
+        GgswCiphertext: GgswCiphertextEntity,
+    {
+        if glwe_input.polynomial_size().0 != ggsw_input.polynomial_size().0 {
+            return Err(Self::PolynomialSizeMismatch);
+        }
+        if glwe_input.glwe_dimension().0 != ggsw_input.glwe_dimension().0 {
+            return Err(Self::GlweDimensionMismatch);
+        }
+        Ok(())
+    }
+}
+
+/// A trait for engines computing the external product between a GLWE ciphertext (of dimension
+/// 1) and a GSW ciphertext.
+///
+/// # Semantics
+///
+/// This [pure](super#operation-semantics) operation generates a GLWE ciphertext vector containing
+/// the result of the external product between a `glwe_input` GLWE ciphertext and a `ggsw_input`
+/// GSW ciphertext.
+///
+/// # Formal Definition
```

Comment:
Here the equations should be in latex (use `\\[ \\]` markers)

Reply:
I will add them.
(To be honest, this part was already there when I started, so maybe this is a work in progress?)

Reply:
May I ask you to check with @damienligier and @ilaria-zama if they have an updated description of it in obsidian? Thanks :smile: